### PR TITLE
FIxes for upcoming ghc version

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: hedgehog/src/Hedgehog/Internal/TH.hs [hunk 1]
```diff
@@ -13,6 +14,9 @@ import           Hedgehog.Internal.Discovery
 import           Hedgehog.Internal.Property
 
 import           Language.Haskell.TH
+#if MIN_VERSION_template_haskell(2, 12, 0)
+                    hiding (TExpQ)
+#endif
 import           Language.Haskell.TH.Syntax
```
Comment (jystic) [inline hedgehog/src/Hedgehog/Internal/TH.hs]: This was lazy on my part, they should have been explicit imports from the beginning (then `CPP` wouldn't be required here.)

Comment (erikd) [inline hedgehog/src/Hedgehog/Internal/TH.hs]: I can make all the import qualified or explicit, either just for this file or for all files. Your choice.

File: hedgehog/src/Hedgehog/Internal/TH.hs [hunk 2]
```diff
@@ -12,8 +12,8 @@ import qualified Data.Ord as Ord
 import           Hedgehog.Internal.Discovery
 import           Hedgehog.Internal.Property
 
-import           Language.Haskell.TH
-import           Language.Haskell.TH.Syntax
+import           Language.Haskell.TH (Exp (..), Q, TExp, location, runIO)
+import           Language.Haskell.TH.Syntax (Loc (..), mkName, unTypeQ, unsafeTExpCoerce)
```
Comment (jystic) [inline hedgehog/src/Hedgehog/Internal/TH.hs]: Just a minor nit, can you remove the space between `Exp` and `(..)` to match the style of the rest of the project (same for `Loc`)

**Comment by jystic** [comment]
👍 after nitpick

File: hedgehog/src/Hedgehog/Internal/TH.hs (hunk 2 above)
Comment (erikd) [inline hedgehog/src/Hedgehog/Internal/TH.hs]: Done!

**Comment by erikd** [comment]
Fixed nitpicks, but I can't merge this.

**Comment by jystic** [comment]
Rebased and merged 🌵

**Pull request closed without merging**